### PR TITLE
Context api, Refactoring massive code into components, adding theme colours

### PR DESCRIPTION
**Pull request by wrt95**
# This is a very large PR - Changes made:
### Context API
Context API has been added to help handling states. The application has become bigger, and we need this to handle all state changes. It works as a global state handler. 

### TextAreaForm
All textAreas has been changed to use the new component TextAreaForm.
    - The reason is that the text area was used so many placed. To make the code smaller and more readable, a component to display the text area inside a form was created. 

### Notification page - ```NotificationsCard.tsx```
Notification page is now displaying the notifications using a Notifications Card component . 
    - The reason is that we might need several notification cards, and to make the code smaller and more readable, a component to display the notification card has been created. 

### FAQ - ```FAQ-Card.tsx``` & ```FAQ-QandAs.ts```
FAQ page is now displaying the FAQ cards using the FAQCard component. 
    - The reason is that we might need several FAQ cards, and to make the code smaller and more readable, a component to display the FAQs card has been created. 
The FAQs get the questions and answers from the file ```FAQ-QandAs.ts```. This file has an object, where question and answers can easily be made. 

### General Style
General Style has been refactored to fit the style from Figma better. 

### Colour theme
Using the Context State handling, there is added a functionality to switch between a light and a dark theme. 
The different colours are set in the file ```ThemeColours.ts```.

### PHOTOS FROM iOS:
(everything tested and works in Android)
Screenshot 2021-02-02 at 16 31 19
Screenshot 2021-02-02 at 16 31 11
Screenshot 2021-02-02 at 16 31 04
Screenshot 2021-02-02 at 16 28 52
Screenshot 2021-02-02 at 16 28 44
Screenshot 2021-02-02 at 16 28 38
Screenshot 2021-02-02 at 16 28 30
Screenshot 2021-02-02 at 16 28 23
Screenshot 2021-02-02 at 16 28 19
Screenshot 2021-02-02 at 16 28 13

### CODE REVIEW

File: ios/SnapThatPhoneApp/Info.plist [hunk 1]
```diff
@@ -87,6 +87,34 @@
...
+		<string>Foundation 2.ttf</string>
+		<string>Ionicons 2.ttf</string>
+		<string>MaterialCommunityIcons 2.ttf</string>
+		<string>MaterialIcons 2.ttf</string>
+		<string>Octicons 2.ttf</string>
+		<string>Roboto 2.ttf</string>
+		<string>Roboto_medium 2.ttf</string>
+		<string>rubicon-icon-font 2.ttf</string>
+		<string>SimpleLineIcons 2.ttf</string>
+		<string>Zocial 2.ttf</string>
```
Comment (calumshep) [inline ios/SnapThatPhoneApp/Info.plist:117]: How come so many new fonts have been added?

File: src/Components/util/TextAreaForm.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+const styles = StyleSheet.create({
+    textArea_container: {
+        backgroundColor: '#D9D9D9',
+        borderRadius: 10,
+        width: 300,
+        marginBottom: 15,
+    },
+    textArea: {
+        width: 300,
+        padding: 10,
```
Comment (calumshep) [inline src/Components/util/TextAreaForm.tsx:46]: Can this padding be increased so the text inside the input is vertically centred?

**Review by calumshep — CHANGES_REQUESTED** [review]

File: ios/SnapThatPhoneApp/Info.plist (hunk 1 above)
Comment (wrt95) [reply ios/SnapThatPhoneApp/Info.plist:117]: For some stupid reason, everything is copied a few times it seems like. Found it on Android, but not iOS. Thanks for letting me know. 
It is sorted now.

File: src/Components/util/TextAreaForm.tsx (hunk 2 above)
Comment (wrt95) [reply src/Components/util/TextAreaForm.tsx]: Fixed

**Review by calumshep — APPROVED** [review]

**Review by sp103 — APPROVED** [review]
Everything looks great, will be nice once this is added to master!

**Comment by rb114-hw** [comment]
Good changes. Nice work!

**Pull request merged**